File: data/EXP01/python/2069/2069.output_2.py
# @lc app=leetcode id=2069 lang=python3
class Robot:
    def __init__(self, width: int, height: int):
        self.width = width
        self.height = height
        self.x = 0
        self.y = 0
        self.direction = "East"
        
    def step(self, num: int) -> None:
        for _ in range(num):
            if self.direction == "North":
                self.y += 1
                if self.y >= self.height:
                    self.y -= 1
                    self.direction = "West"
            elif self.direction == "East":
                self.x += 1
                if self.x >= self.width:
                    self.x -= 1
                    self.direction = "North"
            elif self.direction == "South":
                self.y -= 1
                if self.y < 0:
                    self.y += 1
                    self.direction = "East"
            elif self.direction == "West":
                self.x -= 1
                if self.x < 0:
                    self.x += 1
                    self.direction = "South"
    
    def getPos(self) -> List[int]:
        return [self.x, self.y]
    
    def getDir(self) -> str:
        return self.direction

File: data/EXP01/golang_no_system_var/2069/2069.output_2.go
type Robot struct {
    width, height int
    x, y          int
    direction     string
}

func Constructor(width int, height int) Robot {
    return Robot{
        width:     width,
        height:    height,
        x:         0,
        y:         0,
        direction: "East",
    }
}

func (r *Robot) Step(num int) {
    for i := 0; i < num; i++ {
        newX, newY := r.getNextPosition()
        if r.isOutOfBounds(newX, newY) {
            r.rotate90Degrees()
            continue
        }
        r.x = newX
        r.y = newY
    }
}

func (r *Robot) GetPos() []int {
    return []int{r.x, r.y}
}

func (r *Robot) GetDir() string {
    return r.direction
}

func (r *Robot) getNextPosition() (int, int) {
    newX, newY := r.x, r.y
    if r.direction == "North" {
        newY++
    } else if r.direction == "East" {
        newX++
    } else if r.direction == "South" {
        newY--
    } else if r.direction == "West" {
        newX--
    }
    return newX, newY
}

func (r *Robot) isOutOfBounds(x, y int) bool {
    return x < 0 || y < 0 || x >= r.width || y >= r.height
}

func (r *Robot) rotate90Degrees() {
    if r.direction == "North" {
        r.direction = "West"
    } else if r.direction == "West" {
        r.direction = "South"
    } else if r.direction == "South" {
        r.direction = "East"
    } else if r.direction == "East" {
        r.direction = "North"
    }
}